Tests/AfluentTests/SequenceTests/DeferredSequenceTests.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  DeferredSequenceTests.swift
+//
+//
+//  Created by Annalise Mariottini on 12/20/23.
+//
+
+import Afluent
+import Foundation
+import XCTest
+
+class DeferredSequenceTests: XCTestCase {
```

Comment:
Can you add more tests to assert that Defer enables retrying?

Reply:
I _think_ `testReturnsANewIteratorEachTime` covers this. Could you take a second look and lmk if it doesn't?

Reply:
It sort of does, in that it makes sure a new iterator is created which is the current requirement for retry. However, it doesn't in that it does not explicitly fail and use the `retry` operator, which is what I'm hoping for. As things stand today it won't prove much more, but if and when there's a refactor it might catch something important.

Reply:
In fairness, I haven't done this anywhere else, but I also think that'll be a big motivator to use `Deferred`. You'll drive yourself insane doing too much cross-operator testing.

Reply:
Ok, I think I understand a bit more. I'll add a test that's more explicit.

Reply:
Looks great!